src/codegen/encoding/scale_encoding.rs

```diff
@@ -578,4 +580,40 @@ impl AbiEncoding for ScaleEncoding {
     fn is_packed(&self) -> bool {
         self.packed_encoder
     }
+
+    fn const_encode(&self, args: &[Expression]) -> Option<Vec<u8>> {
+        let mut result = vec![];
+        for arg in args {
+            match arg {
+                Expression::AllocDynamicBytes {
+                    initializer: Some(data),
+                    ty: Type::String,
+                    ..
+                } => result.extend_from_slice(&String::from_utf8(data.to_vec()).unwrap().encode()),
+                Expression::AllocDynamicBytes {
+                    initializer: Some(data),
+                    ty: Type::Slice(_),
+                    ..
+                } => result.extend_from_slice(data),
```

Comment:
That's wrong, if `Type::Slice(ty)` has `ty == Type::Bytes(1)` then it should have the same implementation as `Type::String`.

Reply:
Good point

Reply:
`Type::String` and `Type::DynamicBytes` encode the same way